components/update_user_group_modal/update_user_group_modal.tsx

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import {Modal} from 'react-bootstrap';
+
+import {FormattedMessage} from 'react-intl';
+
+import * as Utils from 'utils/utils.jsx';
+import {CustomGroupPatch, Group} from 'mattermost-redux/types/groups';
+
+import 'components/user_groups_modal/user_groups_modal.scss';
+import './update_user_group_modal.scss';
+import {ModalData} from 'types/actions';
+import Input from 'components/input';
+import {ActionResult} from 'mattermost-redux/types/actions';
+import LocalizedIcon from 'components/localized_icon';
+import {t} from 'utils/i18n';
+
+import SaveButton from 'components/save_button';
+
+export type Props = {
+    onExited: () => void;
+    groupId: string;
+    group: Group;
+    backButtonCallback: () => void;
+    actions: {
+        patchGroup: (groupId: string, group: CustomGroupPatch) => Promise<ActionResult>;
+        openModal: <P>(modalData: ModalData<P>) => void;
+    };
+}
+
+type State = {
+    saving: boolean;
+    show: boolean;
+    name: string;
+    mention: string;
+    hasUpdated: boolean;
+    mentionInputErrorText: string;
+    nameInputErrorText: string;
+    showUnknownError: boolean;
+}
+
+export default class UpdateUserGroupModal extends React.PureComponent<Props, State> {
```

Comment:
nit: can this be rewritten as a functional component?

Reply:
This is now a functional component